columnphysics/icepack_shortwave.F90

```diff
@@ -4815,16 +4813,33 @@ subroutine compute_dEdd_5bd(                          &
                   ks = ext_cff_mss_ice_drc(ns,nmbrad_snicar)
                   ws = ss_alb_ice_drc     (ns,nmbrad_snicar)
                   gs = asm_prm_ice_drc    (ns,nmbrad_snicar)
-               elseif (ceiling(rsnw(ksnow)) - rsnw(ksnow) < 1.0e-3_dbl_kind) then
-                  ! radius = 30 --> nr = 1 in SNICAR table      ! NOTE
-                  nr = ceiling(rsnw(ksnow)) - 30 + 1            ! hardwired min radius = 30
-                  ks = ext_cff_mss_ice_drc(ns,nr)
-                  ws = ss_alb_ice_drc     (ns,nr)
-                  gs = asm_prm_ice_drc    (ns,nr)
-               else ! linear interpolation in rsnw
-                  nr = ceiling(rsnw(ksnow)) - 30 + 1            ! hardwired min radius = 30
-                  delr = (rsnw(ksnow) - floor(rsnw(ksnow))) &   ! hardwired delta radius = 1 in table
-                       / (ceiling(rsnw(ksnow)) - floor(rsnw(ksnow))) ! denom always = 1
+               else
+                  ! check whether rsnw_snicar_chk are integers with delta of one, makes search faster
+                  if (rsnw_snicar_chk == 'unknown') then
+                     rsnw_snicar_chk = 'delta1'
```

Comment:
why not initialize rsnw_snicar_chk = 'delta1' above?  Or assume it's something other than delta1 and rearrange the logic?

Reply:
You only want to do this check once.  So it's initialized to "unknown".  Then on first entry, it decides if it's "delta1" or "mixed" and then these checks are never done again.  This is like a logical (first_call), but we need more than two states.  We could have a first_call logical and then we could have a string that defaults to "mixed" then resets on first_call to "delta1" if it meets the criteria.  Would that be clearer?

Reply:
that would be clearer.  Even better, could this check be moved into an initialization routine somewhere in columnphysics?

Reply:
OK, let me see what I can do.

Reply:
OK, I refactored this part.  I added a new variable, rsnw_datatype, to icepack_shortwave_data.F90, so that variable is set at initialization.  Then the value of that variable can be used to manage the rsnw interpolation logic.

Reply:
This is much clearer now, thank you.